resolwe/flow/views.py

```diff
@@ -500,6 +501,97 @@ def remove_data(self, request, pk=None):
         return Response()
 
 
+class EntityViewSet(CollectionViewSet):
+    """API view for entities."""
+
+    filter_class = EntityFilter
+    serializer_class = EntitySerializer
+
+    queryset = Entity.objects.annotate(
+        latest_date=Max('data__modified')
+    ).prefetch_related(
+        'descriptor_schema',
+        'contributor'
+    ).order_by('-latest_date')
```

Comment:
Same query was quick enough in sample view. I've added prefetch for `data` objects and db index for `modified` field, so now it should be even quicker. I think that we can optimize this when problem will appear.